src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/NodeStepNamePredicate.java

```diff
@@ -52,6 +58,17 @@ public boolean apply(@Nullable FlowNode input) {
         if (input instanceof StepNode) {
             StepDescriptor sd = ((StepNode) input).getDescriptor();
             return (sd != null) ? descriptorId.equals(sd.getId()) : false;
+        } else if (input != null && !(input instanceof FlowStartNode || input instanceof FlowEndNode)) {
+            try {
+                // Workaround for cases where someone is using the latest workflow-api which has StepNode
+                //  but manages to skip the post-2.26 versions of workflow-cps where it a parent of workflow-cps StepNode.
+                // Technically consumers *were* supposed to call ForkScanner#setParallelStartPredicate, but better to be foolproof.
+                Method getDescriptorMethod = input.getClass().getDeclaredMethod("getDescriptor", null);
```

Comment:
Why `getDeclaredMethod`, when this is supposed to have been `public`? And why the non-varargs call? I think you meant just

```java
Method getDescriptorMethod = input.getClass().getMethod("getDescriptor");
```